File: src/CHTL/CHTLNode/ImportNode.java
package CHTL.CHTLNode;

/**
 * Represents an [Import] statement in the AST.
 * This node is processed by the parser itself to load other files,
 * and may not exist in the final AST that goes to the processor.
 */
public class ImportNode implements BaseNode {

    private final String path;

    public ImportNode(String path) {
        this.path = path;
    }

    public String getPath() {
        return path;
    }

    @Override
    public <R> R accept(Visitor<R> visitor) {
        return visitor.visitImportNode(this);
    }

    @Override
    public BaseNode clone() {
        return new ImportNode(path);
    }
}

File: CHTL/CHTLNode/ImportNode.cpp
#include "ImportNode.h"
#include "Visitor.h"

namespace CHTL {

void ImportNode::accept(Visitor& visitor) {
    visitor.visit(*this);
}

std::unique_ptr<BaseNode> ImportNode::clone() const {
    auto node = std::make_unique<ImportNode>();
    node->category = this->category;
    node->type = this->type;
    node->name = this->name;
    node->path = this->path;
    node->alias = this->alias;
    return node;
}

} // namespace CHTL
